### ISSUE TITLE
aio: API doc formatting glitches

### ISSUE THREAD

**Issue by wardbell** [angular/angular#16449]
Odd formatting as seen in this excerpt from a recent PR:

https://pr16446-9b158d7e3a6eeaa6f44fef88b06a787e50a87ddf.ngbuilds.io/api/common/NgIf

![image](https://cloud.githubusercontent.com/assets/129061/25568006/6b0aee40-2dae-11e7-9353-e2691a8de41e.png)

**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Comment by sjtrimble** [comment]
This is happening because the TOC gets inserted right after the title - which is deduced as the h1 element in most pages when there isn't a title. Would this be fixed by adding a title to each document or do we need to change where the TOC gets inserted for the API pages? This is something that needs to be addressed programmatically, not with styles.

Can this be something that doesn't happen automatically, but instead something that the author inserts, as a widget or sorts?

**Comment by sjtrimble** [comment]
@wardbell I'm not sure how to do this, but if we can change it so that IF there is a <header> tag, we put it after that instead of the <h1> tag, that would fix this on the API pages.

**Comment by petebacondarwin** [comment]
> @wardbell I'm not sure how to do this, but if we can change it so that IF there is a <header> tag, we put it after that instead of the <h1> tag, that would fix this on the API pages.

This is in an interesting idea. Another idea would be to change the docViewer so that it doesn't try to insert a ToC if there is already a `<aio-toc>` element in the document already. Then in the API docs we could choose where to place the ToC.

**Comment by petebacondarwin** [comment]
I understand from talking to @wardbell that we can just add `no-toc` to the `<h1>` and then place our own `<aio-toc>` wherever we see fit in the API docs. This works for me. We have already a PR in place to add the `no-toc` so we can decide later if we want to have an embedded ToC in the pages.

**Comment by petebacondarwin** [comment]
I believe this is fixed by ea02073c8429503dda7ff1a10039e92f3238eb88
